gamemodes/horde/gamemode/gadgets/gadget_solar_array.lua

```diff
@@ -1,22 +1,34 @@
 GADGET.PrintName = "Solar Array"
-GADGET.Description = "Drops an armor battery."
+GADGET.Description = "Triggers a pulse that gives armor to nearby players.\nRepairs 20 armor."
 GADGET.Icon = "items/gadgets/solar_array.png"
 GADGET.Duration = 0
-GADGET.Cooldown = 30
+GADGET.Cooldown = 10
 GADGET.Active = true
 GADGET.Params = {
 }
 GADGET.Hooks = {}
 
+
 GADGET.Hooks.Horde_UseActiveGadget = function (ply)
     if CLIENT then return end
     if ply:Horde_GetGadget() ~= "gadget_solar_array" then return end
-    local ent = ents.Create("item_battery")
-    local pos = ply:EyePos()
-    local dir = ply:GetAimVector()
-    local drop_pos = pos + dir * 50
-    drop_pos.z = pos.z + 15
-    ent:SetPos(drop_pos)
-    ent:SetAngles(Angle(0, ply:GetAngles().y, 0))
-    ent:Spawn()
+    local effectdata = EffectData()
+    effectdata:SetOrigin(ply:GetPos())
+    effectdata:SetRadius(225)
+    util.Effect("horde_solar_array", effectdata)
+    ply:EmitSound("horde/player/life_diffuser.ogg", 100, 100, 1, CHAN_AUTO)
+
+    for _, ent in pairs(ents.FindInSphere(ply:GetPos(), 200)) do
+        if ent:IsPlayer() then
+        ent:SetArmor(math.min (ent:GetMaxArmor(), ent:Armor() + 20))
+    elseif ent:IsNPC() and ent:GetClass() ~= "npc_vj_horde_antlion" then
```

Comment:
Fix the indentation on this and it should be good

Reply:
resolved